From fadb55b7d0a5dc6ec36ef97ab7fac11f94b945d9 Mon Sep 17 00:00:00 2001
From: Julie REGAIRAZ <julie.regairaz@intel.com>
Date: Thu, 22 May 2014 19:17:28 +0200
Subject: HS2.0R2: Build OSU client on Android

BZ: 194997

OSU client is linking against libxml2 static library.
Hence it needs symbol defined in "schemas" files
Setting -DLIBXML_SCHEMAS_ENABLED in CFLAGS

Change-Id: I0ddc0a2da624dea48a581df75599a51e2ccd33e9
Category: device enablement
Domain: CWS.WIFI-Common
Origin: internal
Upstream-Candidate: no, proprietary
Signed-off-by: Julie REGAIRAZ <julie.regairaz@intel.com>
---
 Android.mk |   21 +++++++++------------
 1 files changed, 9 insertions(+), 12 deletions(-)

diff --git a/Android.mk b/Android.mk
index b7adff3..c7c1242 100644
--- a/Android.mk
+++ b/Android.mk
@@ -50,32 +50,29 @@ common_SRC_FILES := \
 
 common_C_INCLUDES += \
 	$(LOCAL_PATH)/include
-
-# For the device
-# =====================================================
+# For the host
+# ========================================================
 
 include $(CLEAR_VARS)
-
 LOCAL_SRC_FILES := $(common_SRC_FILES)
 LOCAL_C_INCLUDES += $(common_C_INCLUDES) external/icu4c/common
 LOCAL_SHARED_LIBRARIES += $(common_SHARED_LIBRARIES)
-LOCAL_CFLAGS += -fvisibility=hidden
-
 LOCAL_MODULE:= libxml2
+include $(BUILD_HOST_STATIC_LIBRARY)
 
-include $(BUILD_STATIC_LIBRARY)
-
-
-# For the host
-# ========================================================
+# For the device
+# =====================================================
 
 include $(CLEAR_VARS)
+
 LOCAL_SRC_FILES := $(common_SRC_FILES)
 LOCAL_C_INCLUDES += $(common_C_INCLUDES) external/icu4c/common
 LOCAL_SHARED_LIBRARIES += $(common_SHARED_LIBRARIES)
+LOCAL_CFLAGS += -fvisibility=hidden -DLIBXML_UNICODE_ENABLED -DLIBXML_SCHEMAS_ENABLED -DLIBXML_REGEXP_ENABLED -DLIBXML_PATTERN_ENABLED
+
 LOCAL_MODULE:= libxml2
-include $(BUILD_HOST_STATIC_LIBRARY)
 
+include $(BUILD_STATIC_LIBRARY)
 
 # For the host on build systems with schema validation
 # ========================================================
-- 
1.7.4.1

